Bowling/Bowling/Frames.cs

```diff
@@ -0,0 +1,286 @@
+using System.Text;
+
+namespace BowlingApp
+{
+    public class Frames
+    {
+        public const int frameNumberTotal = 10;   // total number of frames in a bowling game
+        public const int bonusFrame = 10;    // frame where bonus roll could occur
+        private List<Frames> framesCollection = new List<Frames>();  // the collection of Frames objects
+        private int marksLength = 0;  // provides the total length of bowling scores string
+        private int placeCount = 0;  // placeCount will be used as a pointer in our bowling score string
+        private string score = string.Empty;  // string for each scoring mark in the bowling score string
+
+        public int frameNumber { get; set; }  // the frame number for these rolls
+        public int firstRoll { get; set; }  // first roll score
+        public int secondRoll { get; set; }  // second roll score
+        public int bonusRoll { get; set; }  //  bonus roll score
+        public string frameOutcome { get; set; }  // is final frame outcome a strike, spare, or open?
+        public int frameScore { get; set; }  //  total score for this frame
+
+        public List<Frames> CreateFrames(string bowlingMarks)
+        {
+            // get total length of the string entered.  This will help in ensuring our pointer doesn't look past the end of the string
+            marksLength = bowlingMarks.Length;
+
+            // run procedures to create an object from the Frames class for each of the 10 frames and add it to our frames List collection
+            try
+            {
+                // create loop that builds the 10 Frames objects
+                for (int i = 1; i <= frameNumberTotal; i++)
+                {
+                    // since the 10th frame can have potential bonus rolls, only the first 9 frames are contained in the procedure below.
+                    if ((placeCount <= marksLength) && (i != 10))
+                    {
+                        createFrames1To9(i, bowlingMarks);
+                    }
+                    else  // this procedure will handle only marks retrieved from the 10th frame.
+                    {
+                        createFrame10(i, bowlingMarks);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(String.Concat("There was a problem creating the Frames collection: ", ex.Message));
+            }
+
+            return framesCollection;
+        }
+
+        /** Frame 1-9 Creation **/
+        private void createFrames1To9(int frameNumber, string bowlingMarks)
+        {
+            Frames frame = new Frames();
+            frame.frameNumber = frameNumber;
+
+            // grab score at latest placeCount pointer
+            score = bowlingMarks.Substring(placeCount, 1);
+
+            try
+            {
+                //  STRIKE scoring and class property assignments
+                if (score.Equals("X"))
```

Comment:
`"X"` here is what we refer to as a magic variable: it has meaning to only someone who knows the domain we are working in. If you give `"X"` a descriptive name (store it in a variable) I bet it would make the comment for this line redundant.

Reply:
When I see lots of branching logic in a single method my first thought is to refactor it using the [Strategy Pattern](https://www.dofactory.com/net/strategy-design-pattern), but there are lots of other useful patterns on that site so have a look around see if you can refactor this logic out.